src/main/java/org/opentripplanner/middleware/utils/CommandLineProcessor.java

```diff
@@ -0,0 +1,120 @@
+package org.opentripplanner.middleware.utils;
+
+import org.opentripplanner.middleware.recurringjobs.RecurringJob;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+
+import static org.opentripplanner.middleware.utils.ConfigUtils.DEFAULT_ENV;
+
+/**
+ * Parses the command line to extract which jobs and/or services should be started. The only stipulation, is that if a
+ * config file is defined, it must be the first argument defined. e.g.
+ *
+ * /configurations/test/env.yml --endpoints-only yes --recurring-jobs monitor-all-trips-job
+ */
+public class CommandLineProcessor {
+    public static final List<String> END_POINTS_ONLY_FLAGS = List.of("--endpoints-only", "-E");
+    public static final List<String> RECURRING_JOB_FLAGS = List.of("--recurring-jobs", "-R");
+    public static final String END_POINTS_ERROR_MESSAGE = String.format("Invalid value for %s. Use 'yes' or 'no'.", END_POINTS_ONLY_FLAGS);
+    public static final String RECURRING_JOB_ERROR_MESSAGE = "Unknown recurring job: %s. Valid jobs are: %s";
+
+    private final Set<RecurringJob> recurringJobs;
+    private boolean hadEndPoints;
+    private String configFile;
+
+    public CommandLineProcessor() {
+        this.recurringJobs = RecurringJob.getAllRecurringJobs();
+        this.hadEndPoints = true;
+        this.configFile = DEFAULT_ENV;
+    }
+
+    /**
+     * Parse the arguments provided on the command line. To maintain backward compatibility, if a config file is
+     * required, it must be the first argument and no command e.g. -config is NOT required.
+     */
+    public void parseArguments(String[] arguments) {
+        if (arguments.length > 0 && arguments[0].contains(".yml")) {
+            configFile = arguments[0];
+        }
+        Map<String, Set<String>> groupedParams = parseCommandLineArguments(arguments);
+        groupedParams.forEach((flag, commandValues) -> {
+            if (RECURRING_JOB_FLAGS.contains(flag)) {
+                defineRecurringJobs(commandValues);
+            }
```

Comment:
To only enable the endpoints and run no recurring jobs, the command line would read `--recurring-jobs`, and that feels kinda strange. Maybe including a `--recurring-jobs no` or `--recurring-jobs none` in addition to just `--recurring-jobs` would help reduce confusion.

Reply:
I have updated with `--recurring-jobs none`